services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/docker_api.py

```diff
@@ -506,3 +511,39 @@ async def try_to_remove_network(network_name: str) -> None:
             return await network.delete()
         except aiodocker.exceptions.DockerError:
             log.warning("Could not remove network %s", network_name)
+
+
+async def update_scheduler_data_label(scheduler_data: SchedulerData) -> None:
+    async with docker_client() as client:
+        # NOTE: builtin `DockerServices.update` function is very limited.
+        # Using the same pattern but updating labels
+
+        try:
+            # fetch information from service name
+            service_inspect = await client.services.inspect(scheduler_data.service_name)
+            service_version = service_inspect["Version"]["Index"]
+            service_id = service_inspect["ID"]
+            spec = service_inspect["Spec"]
+
+            # allows to use json encodes not available on dict
+            dict_scheduler_data = json.loads(scheduler_data.json())
+            # compose_spec needs to be json encoded
+            dict_scheduler_data["compose_spec"] = json.dumps(
+                dict_scheduler_data["compose_spec"]
+            )
+            label_data = json.dumps(dict_scheduler_data)
+            spec["Labels"][DYNAMIC_SIDECAR_SCHEDULER_DATA_LABEL] = label_data
+
+            await client._query_json(  # pylint: disable=protected-access
+                "services/{service_id}/update".format(service_id=service_id),
+                method="POST",
+                data=json.dumps(clean_map(spec)),
+                params={"version": service_version},
+            )
+        except aiodocker.exceptions.DockerError as e:
+            if e.message != f"service {scheduler_data.service_name} not found":
```

Comment:
Using a message is also quite unreliable way ot determine a state. 
I see that https://aiodocker.readthedocs.io/en/latest/_modules/aiodocker/exceptions.html#DockerError provides a status flag. If those are http status flags, then e.status==404 might be more reliable.

Reply:
I think that adding `e.status!=404` to the current check would be even better.

I don't see how this is less reliable. I am searching for a very specific message, corresponding to a very specific case.